[IMP] Payroll: add necessary contract modules

larm-odoo opened the pull request:
Updating this doc to add a necessary module that must be installed. This was requested on [this project card](https://www.odoo.com/web#id=3781286&cids=3&model=project.task&view_type=form).

This change will go form 15 up to master, as this infomration is necessary in all versions.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/9314).

larm-odoo commented:
Hi @hojo-odoo - this is ready for a peer review. Please note, I only added an important block in the Contract Details tab section of the doc, and I updated issues that appeared on the Linter (not using "in order to" and fixed all instances of "Kanban").

hojo-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -14,7 +14,7 @@ about their position.
 To view the employee contracts, go to the :menuselection:`Payroll app --> Employees --> Contracts`
 from the top menu. All employee contracts, and their current contract status, are displayed in a
-default kanban view. The default view displays both running contracts and contracts that need
+default Kanban view. The default view displays both running contracts and contracts that need
```
```suggestion
Kanban view, by default. The Kanban view displays both running contracts and contracts that require
```

hojo-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,18 @@ Contract details
+   To access the various contract template fields in the :guilabel:`Contract Details` section, the
+   :guilabel:`Salary Configurator` module must be installed.
+
+   To install this module, navigate to the *Apps* application. In the search bar, delete the
```
```suggestion
   To install this module, navigate to the :menuselection:`Apps` application. In the search bar, delete the
```

hojo-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,18 @@ Contract details
+   To install this module, navigate to the *Apps* application. In the search bar, delete the
+   :guilabel:`Apps` filter, then type in :guilabel:`Salary Configurator`.
+
+   Click :guilabel:`Install` on the :guilabel:`Salary Configurator` module, and that module, along
+   with the :guilabel:`Salary Configurator - Holidays` and :guilabel:`Salary Configurator - Payroll`
+   modules install as well. Once the modules are installed, the database reverts to the main
```
```suggestion
   Click :guilabel:`Install` on the :guilabel:`Salary Configurator` module. Doing so installs
   the :guilabel:`Salary Configurator - Holidays` and :guilabel:`Salary Configurator - Payroll`
   modules as well. Once the modules are installed, the database reverts to the main
```

hojo-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,18 @@ Contract details
+   :guilabel:`Salary Configurator` module must be installed.
+
+   To install this module, navigate to the *Apps* application. In the search bar, delete the
+   :guilabel:`Apps` filter, then type in :guilabel:`Salary Configurator`.
```
```suggestion
   :guilabel:`Apps` filter, then type in `Salary Configurator`.
```

hojo-odoo reviewed (approved):
Hey @larm-odoo,

Good job on this update. I'm approving with a few small suggestions.

Thanks,
hojo

larm-odoo commented:
Hi @ksc-odoo - this is ready for a final review! This is similar to the other doc you reviewed- adding the same info to this doc, as it is necessary here as well.

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,17 @@ Contract details
 The contract details section allows for the addition and editing of a contract, and the ability to
```
```suggestion
The :guilabel:`Contract Details` tab allows for the addition and editing of a contract, along with the ability to
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,17 @@ Contract details
 The contract details section allows for the addition and editing of a contract, and the ability to
 send the contract to the employee for approval and signatures.
```
As you indicate, you mention that in the 'Contract Details' tab of a contact form, users can send the contract to the employee for approval and signatures... can you be more specific where in this tab such an action can be performed? Thanks

larm-odoo replied:
I updated htis section- since you must add templates in here, or you cannot create the contract (and will not see the 'send for signature' button either). I updated htis to be more accurate, and specified these are required fields.

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,17 @@ Contract details
 send the contract to the employee for approval and signatures.
 
+.. important::
+   To access the various contract template fields in the :guilabel:`Contract Details` section, the
```
```suggestion
   To access the various contract template fields in the :guilabel:`Contract Details` tab, the
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,17 @@ Contract details
+   To access the various contract template fields in the :guilabel:`Contract Details` section, the
+   *Salary Configurator* module **must** be installed.
+
+   To install this module, navigate to the :menuselection:`Apps` application. In the search bar,
```
```suggestion
   To install this module, navigate to the :menuselection:`Apps` application. In the :guilabel:`Search...` bar,
```

larm-odoo replied:
I edited this sectoin a bit to match another edit Sam made in a different doc that addresses the same issue.

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -89,6 +89,17 @@ Contract details
+   *Salary Configurator* module **must** be installed.
+
+   To install this module, navigate to the :menuselection:`Apps` application. In the search bar,
+   delete the :guilabel:`Apps` filter, then type in `Salary Configurator`.
```
```suggestion
   delete the :guilabel:`Apps` filter, then search for: `Salary Configurator`.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -131,7 +142,7 @@ A pop-up window appears with all the contract details. Modify the fields for the
   - :guilabel:`On Invitation`: Only users selected in this field can sign the contract.
 
-- :guilabel:`Invited Users`: Select the person(s) that can sign the document.
+- :guilabel:`Invited Users`: Select the persons that can sign the document.
```
```suggestion
- :guilabel:`Invited Users`: Select the person (or people) that can sign the document.
```

ksc-odoo reviewed (changes requested):
Hey, @larm-odoo -- just finished my Final Review. Once you address the comments I've left, and implement the necessary adjustments, please tag me again for another look.

Also...

There are other formatting errors in lines 32-84, which are greyed out at the moment...the need for formatting corrections are pretty abundant in that section, so if you'd like to make another PR in the future to apply those adjustments, that would probably benefit the documentation a great deal.

👍

larm-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -14,7 +14,7 @@ about their position.
 from the top menu. All employee contracts, and their current contract status, are displayed in a
-default kanban view. The default view displays both running contracts and contracts that need
+Kanban view, by default. The Kanban view displays both running contracts and contracts that require
 action. Expired and canceled contracts are hidden in the default view.
```
Also- they actually implemented my tiket for chanigng "canceled" to "cancelled"! It was NOT liek that before - they were using the Queen's English. Well, it's the King's English now, I suppose =)

larm-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -25,7 +25,7 @@ action. Expired and canceled contracts are hidden in the default view.
    The list of contracts in the *Payroll* application matches the list of contracts in the
    *Employees* application. The default contracts view in the *Payroll* application displays running
    contracts and contracts needing attention, while the default contracts view in the *Employees*
```
Double checked- and took screenshots to confirm- it is an identical list.

larm-odoo commented:
Thank you @ksc-odoo! I decided to just go ahead and make those formatting edits, so I made sure everything after a colon was lower case, and I added the new icons as well. I think that's all the fomratting issues that were from our new rules. I also made all the requested edits and updates. Ready for another look!

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -3,30 +3,27 @@ Contracts
-default kanban view. The default view displays both running contracts and contracts that need
-action. Expired and canceled contracts are hidden in the default view.
+Kanban view, by default. The Kanban view displays running contracts, contracts that require action,
+expired contracts, and cancelled contracts, all in the default view.
```
```suggestion
expired contracts, and cancelled contracts.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -42,19 +39,19 @@ Required fields
-- :guilabel:`Working Schedule`: Select one of the working schedules from the drop-down menu.
+- :guilabel:`Start Date`: the date the contract starts. Choose a date by clicking on the drop-down
+  menu, navigating to the correct month and year by using the :icon:`fa-chevron-left`
+  :icon:`fa-chevron-right` :guilabel:`(arrow)` icons, then clicking on the :guilabel:`date`.
```
```suggestion
  :icon:`fa-chevron-right` :guilabel:`(arrow)` icons, then clicking on the desired date.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -42,19 +39,19 @@ Required fields
    :align: center
    :alt: New contract form to be filled in when creating a new contract.
 
```
I totally understand that you prefer to break down forms like this into 'Require' and 'Optional' fields...however, I think it would be best if you simply list the fields in order as they appear on the form, then add in the explanation if they are **required**. That way, the reader doesn't feel like they have to jump around, and scroll back and forth between sections, as they familiarize themselves with the form and its fields. So, if you could implement that adjustment, I think it would greatly benefit the overall doc. Thanks! 👍

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -65,41 +62,52 @@ Required fields
-- :guilabel:`Department`: The department the contract applies to.
-- :guilabel:`Job Position`: The specific job position the contract applies to.
-- :guilabel:`Contract Type`: Choose from :guilabel:`CDI`, :guilabel:`CDD`, or :guilabel:`PFI` from
+- :guilabel:`Employee`: name of the employee that the contract applies to.
```
```suggestion
- :guilabel:`Employee`: name of the employee the contract applies to.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -65,41 +62,52 @@ Required fields
+- :guilabel:`Contract Type`: choose from :guilabel:`CDI`, :guilabel:`CDD`, or :guilabel:`PFI` from
   the drop-down menu.
 
   - :guilabel:`CDI` is an open-ended contract with only a start date but no end date.
```
```suggestion
  - :guilabel:`CDI` is an open-ended contract with only a start date, but no end date.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -65,41 +62,52 @@ Required fields
 
-- :guilabel:`End Date`: If the contract has a specific end date, click the drop-down menu, navigate
+- :guilabel:`End Date`: if the contract has a specific end date, click the drop-down menu, navigate
   to the correct month and year using the arrow icons, then click on the date.
```
do you want to implement those same arrow icons that you used previously? that might be a nice touch here.

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -65,41 +62,52 @@ Required fields
+- :guilabel:`HR Responsible`: if there is a specific person in HR that is responsible for the
   contract, select the person from the drop-down menu.
-- :guilabel:`Analytic Account`: This field allows a link between the contract and a specific
+- :guilabel:`Analytic Account`: this field allows a link between the contract and a specific
```
Isn't the 'Analytic Account' field located in the 'Contract Details' tab?
![Screenshot 2024-05-15 111642](https://github.com/odoo/documentation/assets/73958186/74321d41-16f9-4a7c-b028-acd51dcfb152)

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -65,41 +62,52 @@ Required fields
+- :guilabel:`Notes`: the notes field is a text field where any notes for the employee contract can
   be entered for future reference.
 
 Modifying a contract
```
I *believe* there is an 'external link' icon available now -- might be nice to add that in this section below

ksc-odoo replied:
Also, can you be more specific where this external link button should be clicked? At this point, the reader is on a contract form, and if they're in 'edit' mode, they can click the external link on a few different fields... so, I think some elaboration is necessary here.

ksc-odoo replied:
Do you mean on the 'New Contract Document Template' field?

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
-- :guilabel:`Signed Document Workspace`: This is where the signatures are stored. Choose a
+- :guilabel:`Tags`: select any tags associated with the contract.
+- :guilabel:`Signed Document Workspace`: this is where the signatures are stored. Choose a
   pre-configured workspace or create a new one.
```
can you briefly describe how a new one can be created from here? thanks 🙏

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
+- :guilabel:`Signed Document Workspace`: this is where the signatures are stored. Choose a
   pre-configured workspace or create a new one.
-- :guilabel:`Signed Document Tags`: Select or create any tags associated only with the signed
+- :guilabel:`Signed Document Tags`: select or create any tags associated only with the signed
```
```suggestion
- :guilabel:`Signed Document Tags`: select or create any tags that are only associated with the signed
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
   pre-configured workspace or create a new one.
-- :guilabel:`Signed Document Tags`: Select or create any tags associated only with the signed
+- :guilabel:`Signed Document Tags`: select or create any tags associated only with the signed
   contract as opposed to the original unsigned contract.
```
```suggestion
  contract, as opposed to the original unsigned contract.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
   contract as opposed to the original unsigned contract.
-- :guilabel:`Redirect Link`: Enter a redirect link for the employee to access the contract. A
+- :guilabel:`Redirect Link`: enter a redirect link for the employee to access the contract. A
   redirect link takes the user from one URL to another, in this case, to the newly updated contract
   specifically written for them.
```
```suggestion
  redirect link takes the user from one URL to another. In this case, it takes them to the newly-updated contract
  specifically written for them.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
-  must be a PDF. To remove the document, click the :guilabel:`🗑️ (trash can)` icon.
+- :guilabel:`Invited Users`: select the person (or people) that can sign the document.
+- :guilabel:`Document`: the attached document can be replaced by clicking the :icon:`fa-pencil`
+  :guilabel:`(pencil)` icon. A pop-up window appears so another document can be selected for upload.
```
```suggestion
  :guilabel:`(pencil)` icon. A pop-up window appears, so another document can be selected for upload.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
+- :guilabel:`Invited Users`: select the person (or people) that can sign the document.
+- :guilabel:`Document`: the attached document can be replaced by clicking the :icon:`fa-pencil`
+  :guilabel:`(pencil)` icon. A pop-up window appears so another document can be selected for upload.
+  The file must be a PDF. To remove the document, click the :icon:`fa-trash-o` :guilabel:`(trash
```
```suggestion
  The file **must** be a PDF. To remove the document, click the :icon:`fa-trash-o` :guilabel:`(trash
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -118,23 +126,24 @@ A pop-up window appears with all the contract details. Modify the fields for the
+  can)` icon.
 
 Once the edits are complete, click the :guilabel:`Save` button. All the information for the selected
 contract template populates the fields in the :guilabel:`Salary Information` tab. Any additional
```
I'm not seeing any of the mentioned fields in the 'Salary Information' tab on the 15.0 runbot I am using for the review...
![Screenshot 2024-05-15 112704](https://github.com/odoo/documentation/assets/73958186/c847eb2d-3fd3-44cf-8284-2d28b7a97db2)

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -173,19 +182,20 @@ To add a new deduction, click :guilabel:`Add a line`. Type in a description for
 
```
can you please elaborate on where the 'salary attachment' option/field/tab(?) can be located? I am having a hard time locating it.

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -173,19 +182,20 @@ To add a new deduction, click :guilabel:`Add a line`. Type in a description for
 
-- :guilabel:`Attachment of Salary`: Any payments taken out towards something that is *not* child
+- :guilabel:`Attachment of Salary`: any payments taken out towards something that is *not* child
   support. Typically any garnishments such as lawsuit payments, payments toward taxes owed, etc.
```
```suggestion
  support. Typically, any garnishments, such as lawsuit payments, payments toward taxes owed, etc.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -199,7 +209,7 @@ Click on one of the following buttons to send the contract to the employee:
 
-- :guilabel:`Generate Simulation Link`: This option is for Belgian companies only. Clicking this
+- :guilabel:`Generate Simulation Link`: this option is for Belgian companies only. Clicking this
   opens a pop-up window that contains the basic information from the contract as well as a link for
```
```suggestion
  opens a pop-up window that contains the basic information from the contract, as well as a link for
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -199,7 +209,7 @@ Click on one of the following buttons to send the contract to the employee:
-- :guilabel:`Generate Simulation Link`: This option is for Belgian companies only. Clicking this
+- :guilabel:`Generate Simulation Link`: this option is for Belgian companies only. Clicking this
   opens a pop-up window that contains the basic information from the contract as well as a link for
   the contract when using the salary configurator. Click :guilabel:`Send Offer` to send an email to
```
```suggestion
  the contract when using the salary configurator. Click :guilabel:`Send` to send an email to
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -199,7 +209,7 @@ Click on one of the following buttons to send the contract to the employee:
+- :guilabel:`Generate Simulation Link`: this option is for Belgian companies only. Clicking this
   opens a pop-up window that contains the basic information from the contract as well as a link for
   the contract when using the salary configurator. Click :guilabel:`Send Offer` to send an email to
   the employee so they can sign the contract.
```
```suggestion
  the employee, so they can sign the contract.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -212,7 +222,7 @@ Click on one of the following buttons to send the contract to the employee:
      In order to send a contract using the :guilabel:`Generate Simulation Link`, there must be a
      signature field in the contract PDF being sent to the employee so they can sign it.
```
```suggestion
     signature field in the contract PDF being sent to the employee, so they can sign it.
```

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -212,7 +222,7 @@ Click on one of the following buttons to send the contract to the employee:
      signature field in the contract PDF being sent to the employee so they can sign it.
 
-- :guilabel:`Signature Request`: Click this and a pop-up window appears where an email can be typed
+- :guilabel:`Signature Request`: click this and a pop-up window appears where an email can be typed
```
```suggestion
- :guilabel:`Signature Request`: clicking this reveals a pop-up window, where an email can be typed
```

ksc-odoo reviewed (changes requested):
hey @larm-odoo --- almost there. please address the comments I've left, implement the necessary adjustments, and tag again when you think it's ready for another look. Thanks! 👍

larm-odoo commented:
Hi @ksc-odoo - thank you again for your very thorough review. I want to take a moment to apologize, because I honestly couldn't understand why my info was off or wrong- and you had so many correctoins to do. This PR was written to only add the needed module info (it was requested ot be added from a product owner- I know you saw the other docs I added this to, because they all had the same info in them).

My intention was just to add that small part to this doc- a simple 2 point doc. So I did not go in and do all the updates as I normally would. I feel I should have been much more clear on that- because I could have done the module doc (for 2 pts) and then a 3 point doc to update the instrucitons (I al slowly learning how to gamify with points- it's not in my nature, so this is not easy for me, lol).

Also, when I originally wrote the contracts doc, it was for 14- then they were ported up to master. This year, I decided to do all the 17 updates FIRST, then go back and do 16 updates. Only after that would I do any 15 updates, since that version was going to go away soon, and most likely not beofre I did all the updates for the newer 2 versions.

So I just wanted to explain myself because I felt SO BAD that you had all these comments- and I honestly didn't know why things/images looked differet than the runbots I had originaly used- since I was only focusing on adding that admonition on the module. So, thank you for being so thorough- and please accept both my explanation and aplogy for all the work you did editing this doc- it was not my intention to have you do so much work!

larm-odoo commented:
Oh, and this is ready for another look @ksc-odoo =D

ksc-odoo commented on content/applications/hr/payroll/contracts.rst:
```diff
@@ -221,5 +211,51 @@ Click on one of the following buttons to send the contract to the employee:
+
+  - :guilabel:`Attachment of Salary`: any payments taken out towards something that is *not* child
+    support. Typically any garnishments such as lawsuit payments, payments toward taxes owed, etc.
+  - :guilabel:`Assignment of Salary`: any deduction that is not required but voluntary, such as a
```
```suggestion
  - :guilabel:`Assignment of Salary`: any deduction that is not required, but voluntary, such as a
```

ksc-odoo reviewed (approved):
Okie dokie, @larm-odoo -- just finished giving this another look -- and I'm approving now 👍 nice job on all the revisions. You're *jusssst* about at the finish line. Just have a handful of minor suggestions that require your attention. But, once you implement all the necessary changes, you can feel free to tag this for Tech Review. Thanks again for all your hard work on this one 👏

larm-odoo commented:
Hi @samueljlieber  - this is ready for a tech review. Thanks!

samueljlieber reviewed (approved):
This looks great to me @larm-odoo, thanks for your hard work on this improvement to Contracts!
...
@robodoo r+
